src/components/SearchInput/index.jsx

```diff
@@ -1,19 +1,66 @@
-import { useRef, useState } from "react";
+import { useRef, useState, useEffect } from "react";
+import axios from "axios";
+import useUserInputStore from "../../store/store";
 
 function SearchInput() {
-  const [query, setQuery] = useState("");
+  const { userInput, setUserInput } = useUserInputStore();
+  const [autoCompletions, setAutoCompletions] = useState([]);
+  const [selectedItemIndex, setSelectedItemIndex] = useState(-1);
   const inputRef = useRef(null);
 
-  function handleQueryChange(event) {
-    setQuery(event.target.value);
+  function handleUserInputChange(event) {
+    setUserInput(event.target.value);
   }
 
-  function handleSubmit(event) {
-    event.preventDefault();
+  useEffect(() => {
+    async function getAutoCompletions(userInput) {
+      const baseURL = "http://localhost:3000";
+
+      try {
+        const response = await axios.get(`${baseURL  }/auto-completions`, {
+          params: { userInput },
+        });
+        const {data} = response;
```

Comment:
```suggestion
        const { data } = response;
```
띄어쓰기가 적용되지 않은 것 같습니다.

Reply:
꼼꼼하지 못했네요! 바로 수정하겠습니다!